app/file_sender.py

```diff
@@ -1,36 +1,129 @@
+    'D_FD_HC2W': 'Continuation Questionnaire pack for Wales (Welsh)',
+    'D_FD_HC4': 'Continuation Questionnaire pack for Northern Ireland (English)',
+    'D_FD_I1': 'Individual Questionnaire pack for England',
+    'D_FD_I2': 'Individual Questionnaire pack for Wales (English)',
+    'D_FD_I2W': 'Individual Questionnaire pack for Wales (Welsh)',
+    'D_FD_I4': 'Individual Questionnaire pack for Northern Ireland (English)',
+    'D_CCS_CH1': 'CCS Interviewer Household Questionnaire for England and Wales',
+    'D_CCS_CH2W': 'CCS Interviewer Household Questionnaire for Wales (Welsh)',
+    'D_CCS_CHP1': 'CCS Postback Questionnaire for England and Wales (English)',
+    'D_CCS_CHP2W': 'CCS Postback Questionnaire for Wales (Welsh)',
+    'D_CCS_CCP1': 'CCS Postback Continuation Questionnaire for England & Wales',
+    'D_CCS_CCP2W': 'CCS Postback Continuation Questionnaire for Wales (Welsh)',
+    'D_CCS_CCE1': 'CCS Interviewer CE Manager for England & Wales (English)',
+    'D_CCS_CCE2W': 'CCS Interviewer CE Manager for Wales (Welsh)',
+    'P_IC_ICL1': 'Initial contact letter households - England',
+    'P_IC_ICL2': 'Initial contact letter households - Wales',
+    'P_IC_ICL4': 'Initial contact letter households - Northern Ireland',
+    'P_IC_H1': 'Initial contact questionnaire households - England',
+    'P_IC_H2': 'Initial contact questionnaire households - Wales',
+    'P_IC_H4': 'Initial contact questionnaire households - Northern Ireland'
+}
+
+
 logger = wrap_logger(logging.getLogger(__name__))
 
 
-def process_complete_file(file: Path):
-    # TODO Encrypt, create manifest, send to sftp, delete partial file
-    logger.info(f'doing the file stuff on {file}')
+def process_complete_file(file: Path, pack_code):
+    message = file.read_text()
+    # First encrypt the file and write it to encrypted directory
+    logger.info('Encrypting file', file_name=file.name)
+    encrypted_message = pgp_encrypt_message(message)
+    filename = f'{pack_code}_{datetime.utcnow().strftime("%Y-%m-%dT%H-%M-%S")}'
+    print_file = Config.ENCRYPTED_FILES_DIRECTORY.joinpath(f'{filename}.csv')
+    logger.info('Writing encrypted file', file_name=print_file.name)
+    with open(print_file, 'w') as encrypted_file:
+        encrypted_file.write(encrypted_message)
+
+    # Create the manifest
+    manifest_file = Config.ENCRYPTED_FILES_DIRECTORY.joinpath(f'{filename}.manifest')
+    logger.info('Creating manifest file', manifest_file=manifest_file.name)
+    generate_manifest_file(manifest_file, print_file, pack_code)
+    file_paths = [print_file, manifest_file]
+
+    # Send files to sftp
+    logger.info('Sending files to SFTP', file_paths=file_paths)
+    copy_files_to_sftp(file_paths)
+
+    # Move files to sent directory
+    logger.info('Moving to sent files directory', file_paths=file_paths)
+    for sent_file in file_paths:
+        sent_file.replace(Config.SENT_FILES_DIRECTORY.joinpath(sent_file.name))
```

Comment:
Do we need to keep the sent files locally after they're finished with?

Reply:
I guess once they've been uploaded to the bucket the files can be deleted/shredded